[CLIENT] [MERGE] add Player sprite and movement

MarcPIster opened the pull request:
# Motivation:

We need a moving and nice looking player (starship) in our game. 

# Modifications:

Added all sprites we got from Epitech + sprite for the player.  Also added movement system that's feels not laggy.

# Result:

We now can move a player in our game

wernerramon committed:
[CLIENT] [ADD] added basic movement to player

MarcPIster committed:
[CLIENT] [FIX] add and fix mouvment speed

Motivation:
Player should have an adjustable movementspeed

Modifications:
add _speed to player

Result:
We can now move faster

MarcPIster committed:
[CLIENT] [MODIFY] change movment system

Motivation:
Player should have a nice feeling movment
Modifications:
add velocity to the movement system

Result:
Player can now move smoothly

MarcPIster committed:
[CLIENT] [MODIFY] removed angle for top and bot

Motivation:
Player should have the movment from rtype
Modifications:
remove turning if flying top or botside

Result:
Player can just fly in two directions

MarcPIster committed:
[CLIENT] [MODIFY] removed angle for top and bot

Motivation:
Player should have the movment from rtype
Modifications:
remove turning if flying to left side

Result:
Player will always look at the right side

MarcPIster committed:
[CLIENT] [ADD] added bouncing to player RTB-21

Motivation:
We doesnt want to fly out of the screen
Modifications:
added check if we hit screen boarder and let us bounce back

Result:
We will now slightly bounce if we hit the end of screen

JosefineM2000 commented on Client/CMakeLists.txt:
```diff
@@ -11,7 +11,7 @@ FetchContent_MakeAvailable(SFML)
 find_package(OpenGL REQUIRED OPTIONAL_COMPONENTS OpenGL)
 include_directories(${OPENGL_INCLUDE_DIR})
-add_executable(r-type_client src/main.cpp)
+add_executable(r-type_client src/main.cpp src/Player.cpp src/Player.h)
```
header files should be named .hpp

JosefineM2000 commented on Client/src/Player.cpp:
```diff
@@ -0,0 +1,76 @@
+  setPosition(_position.x, _position.y);
+  _body.setTexture(&_texture);
+  _speed = 10;
+  _velocity = sf::Vector2f(0,0);
```
space missing: (0, 0)

JosefineM2000 commented on Client/src/Player.cpp:
```diff
@@ -0,0 +1,76 @@
+  switch (event.key.code)
+  {
+    case sf::Keyboard::W:
+      move(sf::Vector2f(0,-1));
```
for this and the 3 following: space missing (x, y)

JosefineM2000 commented on Client/src/Player.cpp:
```diff
@@ -0,0 +1,76 @@
+    case sf::Keyboard::A:
+      move(sf::Vector2f(-1,0));
+      break;
+    default:
```
no need for default case if it does nothing

JosefineM2000 commented on Client/src/Player.cpp:
```diff
@@ -0,0 +1,76 @@
+  if (_position.x <= _body.getSize().x / 2) {
+    _velocity.x = 1;
+  }
+  else if (_position.x + _body.getSize().x >= windowSize.x +  _body.getSize().x / 2) {
```
before last body get size: two spaces. same in second else if

JosefineM2000 commented on Client/src/Player.h:
```diff
@@ -0,0 +1,27 @@
+
+class Player {
+ private:
+  sf::Sprite _sprite;
```
never name any variable starting with underscore. Also, data members in class start with 'm_'. Please adjust for all class variables

JosefineM2000 commented on Client/src/Player.h:
```diff
@@ -0,0 +1,27 @@
+#include <SFML/Graphics.hpp>
+#include <iostream>
+
+class Player {
```
order of groups: first public, then protected (in this case not used) and after that private

JosefineM2000 commented on Client/src/Player.h:
```diff
@@ -0,0 +1,27 @@
+  float _angle;
+
+ public:
+  Player(std::string sprite_path);
```
method parameters start with "t_" and follow the snake_case

JosefineM2000 commented on Client/src/main.cpp:
```diff
@@ -1,17 +1,23 @@
-
+  Player player("./sprites/starship.png");
+  sf::RenderWindow window(sf::VideoMode(800, 1400), "R-Type Epitech");
+  sf::Vector2f windowSize =sf::Vector2f (window.getSize().x, window.getSize().y);
```
space missing after =

JosefineM2000 commented on Client/src/main.cpp:
```diff
@@ -1,17 +1,23 @@
+  sf::RenderWindow window(sf::VideoMode(800, 1400), "R-Type Epitech");
+  sf::Vector2f windowSize =sf::Vector2f (window.getSize().x, window.getSize().y);
+  window.setFramerateLimit(60);
   while (window.isOpen()) {
```
I personnally would prefer an empty line above the while loop

JosefineM2000 commented on Client/src/Player.h:
```diff
@@ -0,0 +1,27 @@
+#ifndef R_TYPE_CLIENT_PLAYER_H
```
ends with _HPP instead of _H

JosefineM2000 commented on Client/src/Player.cpp:
```diff
@@ -0,0 +1,76 @@
+  _body.setPosition(_position);
+  if (_velocity.x != 0 || _velocity.y != 0)
+    _velocity *= 0.99f;
+}
```
each file should end with an empty line

JosefineM2000 commented on Client/src/main.cpp:
```diff
@@ -1,17 +1,23 @@
 #include <iostream>
-
+#include "Player.h"
 #include "SFML/Graphics.hpp"
```
pay attention on sorting of imports -> own project files are last group

JosefineM2000 reviewed (changes requested):
Please read the coding style rules

MarcPIster committed:
[CLIENT] [CLEAN] adjust code after review RTB-21

Motivation:
We want to have a clean consisting coding-style
Modifications:
changed h -> hpp, and much more

Result:
We now have clean code

MarcPIster committed:
[CLIENT] [CLEAN] removed include

Motivation:
We want to have a clean consisting coding-style
Modifications:
removed unnecessary includes from main.ccp

Result:
We now have clean code

JosefineM2000 commented on Client/src/Player.hpp:
```diff
@@ -0,0 +1,27 @@
+#define R_TYPE_CLIENT_PLAYER_HPP
+
+#include <SFML/Graphics.hpp>
+#include <iostream>
```
wrong include order: please refer to guidelines

JosefineM2000 reviewed (changes requested):
Please change variable names according to comment + order of groups/imports. And pay attention to the suggestions from cpplint
